src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs

```diff
@@ -39,10 +33,23 @@ internal static class DelegateProxyGenerator
                 }
             }
 
-            var eventRaiser = new DelegateCallInterceptedEventRaiser(fakeCallProcessorProvider, invokeMethod, typeOfProxy);
+            // You'll need distinct `options` per delegate type. Specify additional options as needed:
+            var options = new ProxyGenerationOptions();
+            options.AddDelegateTypeMixin(typeOfProxy);
+
+            // Then use it to generate a proxy. That proxy will have an `Invoke` method with the specified
+            // delegate type's signature. Proxy type, base class, interceptors etc. are chosen as usual:
+            var delegateProxyInterceptor = new DelegateProxyInterceptor(fakeCallProcessorProvider);
+            var proxy = ProxyGenerator.CreateClassProxy(typeof(object), options, delegateProxyInterceptor);
+
+            // Either call `Delegate.CreateDelegate` for the generated `Invoke` method yourself, or use a

```

Comment:
these are comments from Team Castle.Core. Clean 'em up, since, y'know, we actually picked an action.